eox_core/api/v1/tests/integration/test_views.py

```diff
@@ -370,6 +431,595 @@ def test_update_user_in_tenant_user_not_found(self, param: str, value: str) -> N
+        ("username", {"non_field_errors": ["Email or username needed"]}),
+    )
+    @ddt.unpack
+    def test_create_enrollment_missing_required_fields(self, param: str, error: list | dict) -> None:
+        """
+        Test creating a enrollment with missing required fields.
+
+        Open edX definitions tested:
+        - `check_edxapp_enrollment_is_valid`
+
+        Expected result:
+        - The status code is 400.
+        - The response contains a message about the missing field.
+        """
+        user_data = next(FAKE_USER_DATA)
+        self.create_user(self.tenant_x, user_data)
+        enrollment_data = {
+            "username": user_data["username"],
+            "mode": self.mode,
+            "course_id": self.course_id,
+        }
+        enrollment_data.pop(param)
+
+        response = self.create_enrollment(self.tenant_x, enrollment_data)
+
+        response_data = response.json()
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+        self.assertEqual(response_data, error)
+
+    @ddt.data("email", "username")
+    def test_force_create_enrollment_success(self, param: str) -> None:
+        """
+        Test force creating a enrollment with a valid user, course and mode in a tenant.
+
+        Open edX definitions tested:
+        - `get_edxapp_user`
+        - `create_enrollment`
+        - `check_edxapp_account_conflicts`
+        - `check_edxapp_enrollment_is_valid`
+
+        Expected result:
+        - The status code is 200.
+        - The enrollment is created successfully in the tenant with the provided data.
+        """
+        user_data = next(FAKE_USER_DATA)
+        self.create_user(self.tenant_x, user_data)
+        enrollment_data = {
+            param: user_data[param],
+            "course_id": self.course_id,
+            "mode": self.mode,
+            "force": True,
+        }
+
+        response = self.create_enrollment(self.tenant_x, enrollment_data)
+
+        response_data = response.json()
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(response_data["username"], user_data["username"])
+        self.assertEqual(response_data["mode"], enrollment_data["mode"])
+        self.assertEqual(response_data["course_id"], enrollment_data["course_id"])
+        self.assertTrue(response_data["is_active"])
+        self.assertIn("created", response_data)
+
+    @ddt.data("email", "username")
+    def test_create_valid_course_mode_invalid_user(self, param: str) -> None:
+        """
+        Test creating a enrollment with a valid course, valid mode, and a non-existent user in a tenant.
+
+        Open edX definitions tested:
+        - `check_edxapp_enrollment_is_valid`
+
+        Expected result:
+        - The status code is 400.
+        - The response contains an error message about the user not found.
+        - The enrollment is not created in the tenant.
```

Comment:
Item 7 of the [integration tests guidance document](https://internal.docs.edunext.co/en/latest/internal/Teams/Descontrol/Internal_processes/integration_tests_in_plugins/integration_tests_with_openedx_components.html?highlight=integration%20tests#pruebas-de-apis) says we should also check the application's state after each update. I don't know if we did that for the users API, though, not for every test. Do you think we should do that here, or did we decide not to check explicitly because of performance issues?

Reply:
It occurs to me that we can query that the enrollment was created by doing a `self.get_enrollment()` and verify that code **200** or **404** is returned, what do you say?

Reply:
Yes, that would align with what we agreed on in the document.

Reply:
I updated the test of enrollment's API: https://github.com/eduNEXT/eox-core/commit/69692fa07ef30e5dffdc668ef586b731ab72f569
> Do you think we should consider updating the user's API as well as the next step? I think we missed doing those checks there.

Yes, we should also update the user's API tests. I will create a task for it.